pkg/controller/controller.go

```diff
@@ -317,6 +381,46 @@ func (p *csiProvisioner) Provision(options controller.VolumeOptions) (*v1.Persis
 			RequiredBytes: int64(volSizeBytes),
 		},
 	}
+
+	if checkSnapshotSupport {
+		snapshotObj, err := p.snapshotClient.VolumesnapshotV1alpha1().VolumeSnapshots(options.PVC.Namespace).Get(options.PVC.Spec.DataSource.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("error get snapshot %s from api server: %v", options.PVC.Spec.DataSource.Name, err)
+		}
+		glog.V(5).Infof("VolumeSnapshot %+v", snapshotObj)
+
+		snapContentObj, err := p.snapshotClient.VolumesnapshotV1alpha1().VolumeSnapshotContents().Get(snapshotObj.Spec.SnapshotContentName, metav1.GetOptions{})
```

Comment:
Maybe we can check VolumeSnapshot status to see whether it is ready?

Reply:
Good point

Reply:
Done